### ISSUE TITLE
scheduler and optimizer order

### ISSUE THREAD

**Issue by RitaHT** [MIC-DKFZ/nnUNet#2743]
When I use default SGD and PolyLR everything goes without error, (added get_last_lr() to polylr.py)
But when I tried to use torch.optim.AdamW and custom warm up cosine, it will prompt     

> 2025-03-18 14:39:35.655314: Unable to plot network architecture: nnUNet_compile is enabled!
            /home/tans3/miniconda3/envs/nnunet_frame_neue/lib/python3.10/site-packages/torch/optim/lr_scheduler.py:224: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
            warnings.warn(
            /home/tans3/miniconda3/envs/nnunet_frame_neue/lib/python3.10/site-packages/torch/optim/lr_scheduler.py:240: UserWarning: The epoch parameter in `scheduler.step()` was not necessary and is being deprecated where possible. Please use `scheduler.step()` to step the scheduler. During the deprecation, if epoch is different from None, the closed form is used instead of the new chainable form, where available. Please open an issue if you are unable to replicate your use case: https://github.com/pytorch/pytorch/issues/new/choose.
            warnings.warn(EPOCH_DEPRECATION_WARNING, UserWarning)

I wonder even for default, shouldn't it follow scheduler, then optimizer? Why can one prompt but another did not?

```
def run_training(self):
        self.on_train_start()

        for epoch in range(self.current_epoch, self.num_epochs):
            self.on_epoch_start()

            self.on_train_epoch_start() #lr_scheduler.step(self.current_epoch)
            train_outputs = []
            for batch_id in range(self.num_iterations_per_epoch):
                train_outputs.append(self.train_step(next(self.dataloader_train))) #optimizer.step()
            self.on_train_epoch_end(train_outputs)

            with torch.no_grad():
                self.on_validation_epoch_start()
                val_outputs = []
                for batch_id in range(self.num_val_iterations_per_epoch):
                    val_outputs.append(self.validation_step(next(self.dataloader_val)))
                self.on_validation_epoch_end(val_outputs)

            self.on_epoch_end()

        self.on_train_end()
```

Thank you very much!

**Comment by constantinulrich** [comment]
Hi, sorry for the late reply.
Is this still an issue? As I read the message above its only a warning? 
Best
Constantin

**Comment by constantinulrich** [comment]
The scheduler sets the learning rate depending on the epoch, which is done once for the whole epoch (250 optimizer steps)
